File: NumberOfDiscIntersections/NumberOfDiscIntersections.swift
import Foundation
import Glibc

public func solution(_ A : inout [Int]) -> Int {
    let size = A.count
    var result = 0

    for i in 0..<size {
        let a = A[i]

        if i + a >= size - 1 {
            result += size - 1 - i
        } else {
            result += a
            let fromJ = i + a + 1

            for j in fromJ..<size {
                if A[j] > j - fromJ {
                    result += 1
                }
            }
        }

        if result > 10_000_000 {
            return -1
        }
    }

    return result
}

File: NumberOfDiscIntersections/NumberOfDiscIntersections.js
function solution(A) {
    const size = A.length;
    let result = 0;

    for (let i = 0; i < size; i++) {
        const a = A[i];

        if (i + a >= size - 1) {
            result += size - 1 - i;
        } else {
            result += a;
            const fromJ = i + a + 1;

            for (let j = fromJ; j < size; j++) {
                if (A[j] > j - fromJ) {
                    result += 1;
                }
            }
        }

        if (result > 10000000) {
            return -1;
        }
    }

    return result;
}